### PR TITLE
Added unittests for common.py

### PR DESCRIPTION
**Pull request by ThomasJRyan**
I wanted a place to start with contributing so I went with writing some unittests for `common.py`. It has a few gaps as I don't quite yet know how to test them. But with some more time I'll figure it out

### CODE REVIEW

File: tests/test_common.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+    def test_write_class_array(self, genie_class: GenieClass):
+        with pytest.raises(NotImplementedError):
+            genie_class.write_class(GenieClass(), Version.UNDEFINED)
+
+
+class TestByteHandler:
+    @pytest.fixture
+    def byte_handler(self):
+        data = memoryview(b''.join([int.to_bytes(i) for i in range(256)]))
```
Comment (HSZemi) [inline tests/test_common.py]: Using this data makes all of the resulting numbers (except in the `read_int_8_…` tests) very hard to understand/make sense of the numbers. Maybe ditch the fixture for setting up sensible data in each test?

File: tests/test_common.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+            genie_class.to_bytes(Version.UNDEFINED)
+
+    def test_write_debug_string(self, genie_class: GenieClass):
+        assert genie_class.write_debug_string("foobar") == b'`\n\x06\x00foobar'
+
+    def test_write_string(self, genie_class: GenieClass):
+        assert genie_class.write_string(6, "foobar") == b'foobar'
+
+    def test_write_int_8(self, genie_class: GenieClass):
+        assert genie_class.write_int_8(10) == b'\n'
```
Comment (HSZemi) [inline tests/test_common.py]: I don't think it's obvious that `\n` is `10`. Maybe use `\x0a` instead? In the other tests, too.

**Review by HSZemi — CHANGES_REQUESTED** [review]
Here's my initial review. Thanks for working on this!

File: tests/test_common.py (hunk 1 above)
Comment (ThomasJRyan) [reply tests/test_common.py]: I don't disagree with this sentiment. Admittedly this is my first go with `pytest` (I've typically used `unittest` for all my projects) and I was mostly trying to get a feel for it and how fixtures work.

Let me see if I can put together something more in-line with what you're looking for

File: tests/test_common.py (hunk 2 above)
Comment (ThomasJRyan) [reply tests/test_common.py]: Fair enough. ~~Though if I address the first comment then this one becomes obsolete I believe~~ Nevermind, different classes of tests

File: tests/test_common.py (hunk 1 above)
Comment (ThomasJRyan) [reply tests/test_common.py]: Alright, I kept the fixtures, but gave them easier to read values. I wanted to use an actual `.dat` file for these tests, but given how they're at the absolute core of the library it seemed best to use dummy data. Further tests are going to use proper `.dat` files where possible

File: tests/test_common.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+    def test_from_bytes_with_count(self, genie_class: GenieClass):
+        with pytest.raises(NotImplementedError):
+            genie_class.from_bytes_with_count(ByteHandler(b''), 0)
+
+    def test_to_bytes(self, genie_class: GenieClass):
+        with pytest.raises(NotImplementedError):
+            genie_class.to_bytes(Version.UNDEFINED)
+
+    def test_write_debug_string(self, genie_class: GenieClass):
+        assert genie_class.write_debug_string("foobar") == b'`\x0A\x06\x00foobar'
```
Comment (HSZemi) [inline tests/test_common.py]: This backtick looks a bit sus :eyes:

Comment (ThomasJRyan) [reply tests/test_common.py]: Genuinely impressed you caught that. Fixed it up

**Review by HSZemi — APPROVED** [review]
Looks good!

**Pull request merged**